### PR TITLE
[Feature][MasterServer] add cache for workGroup and schedule #6987

### PR DESCRIPTION
**Pull request by zwZjut**
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/mapper/ScheduleMapper.java [hunk 1]
```diff
@@ -67,6 +72,12 @@
      * @param processDefinitionCode processDefinitionCode
      * @return schedule list
      */
+    @Cacheable(sync = true)
     List<Schedule> queryReleaseSchedulerListByProcessDefinitionCode(@Param("processDefinitionCode") long processDefinitionCode);
 
+    @CacheEvict(key = "#entity.processDefinitionCode")
```
Comment (caishunfeng) [inline dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/mapper/ScheduleMapper.java:79]: SpringEL is better, but now the `CacheEvictAspect.java ` can not support it. So can you try to solve SpringEL support?

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/mapper/WorkerGroupMapper.java [hunk 2]
```diff
@@ -23,19 +23,34 @@
...
  * worker group mapper interface
  */
+@CacheConfig(cacheNames = "workerGroup", keyGenerator = "cacheKeyGenerator")
 public interface WorkerGroupMapper extends BaseMapper<WorkerGroup> {
 
     /**
      * query all worker group
      * @return worker group list
      */
+    @Cacheable(sync = true, key = "'-'")
```
Comment (caishunfeng) [inline dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/mapper/WorkerGroupMapper.java]: is better to set key like 'all' ?

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/aspect/CacheEvictAspect.java [hunk 3]
```diff
@@ -99,4 +113,29 @@ private CacheType getCacheType(CacheConfig cacheConfig, CacheEvict cacheEvict) {
         }
         return null;
     }
+
+    private String parseKey(String key, List<String> paramNameList, List<Object> paramList) {
+        SpelExpressionParser spelParser = new SpelExpressionParser();
+        //将方法的参数名和参数值一一对应的放入上下文中
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/aspect/CacheEvictAspect.java]: Please use english for it.

**Review by caishunfeng — COMMENTED** [review]
LGTM

**Review by lenboo — APPROVED** [review]
+1

**Comment by caishunfeng** [comment]
relative issue: #6987

**Pull request merged**